metaflow/plugins/kfp/kfp_s3_sensor.py

```diff
@@ -0,0 +1,58 @@
+"""
+This function is called within the s3_sensor_op running container.
+(1) It decodes the formatter function's code and runs it to obtain a final S3 path
+(2) It splits the formatted path into an S3 bucket and key 
+(3) It polls for an object with the specified bucket and key until timeout
+"""
+
+
+def wait_for_s3_path(
+    path: str,
+    timeout_seconds: int,
+    polling_interval_seconds: int,
+    func_code_encoded: str,
+    flow_parameters_json: str,
+) -> None:
+    import boto3
+    import botocore
+    import base64
+    import marshal
+    import json
+    import time
+    from datetime import date
+    from typing import Tuple
+
+    def split_s3_path(path: str) -> Tuple[str, str]:
+        path = path.replace("s3://", "")
+        bucket, key = path.split("/", 1)
+        return bucket, key
+
+    flow_parameters_json = json.loads(flow_parameters_json)
+
+    func_code = marshal.loads(base64.b64decode(func_code_encoded))
+
+    def formatter(key: str, flow_parameters_json: dict) -> str:
+        pass
+
+    formatter.__code__ = func_code
+    path = formatter(path, flow_parameters_json)
```

Comment:
Just to confirm my understanding if the user doesn't provide the `formatter` argument it'll gracefully fall back to the `identity_formatter` method below which just returns and unmodified `path` string?

Reply:
That's correct :)